Source/JavaScriptCore/runtime/PureNaN.h

```diff
@@ -71,16 +71,21 @@ namespace JSC {
 // the docs appear to imply that quiet_NaN could even return a double with the
 // signaling bit set on hardware that doesn't do signaling. That would probably
 // never happen, but it's healthy to be paranoid.
-static constexpr uint64_t PNaNAsBits { 0x7ff8000000000000ll };
+static constexpr uint64_t PNaNAsBits { 0x7ff8000000000000ULL };
```

Comment:
Can we add some static_asserts for cases where we're expecting to not need to purifyNaN e.g.

```
static_assert(PNaNAsBits == std::bit_cast<uint64_t>(numeric_limits<double>::quiet_NaN()));
static_assert(!isImpureNaN(std::bit_cast<uint64_t>(-PNaN)));
```

Reply:
I think
```
static_assert(PNaNAsBits == std::bit_cast<uint64_t>(numeric_limits<double>::quiet_NaN()));
```

is not guaranteed (I think it is guaranteed that it is impure NaN, but we are just picking one of pure NaN as PNaN). But on the other hand, 

``
static_assert(!isImpureNaN(std::bit_cast<uint64_t>(-PNaN)));
```
this is right.